src/sdl2/ttf/context.rs

```diff
@@ -82,52 +88,25 @@ impl Sdl2TtfContext {
 
 /// Returns the version of the dynamically linked `SDL_TTF` library
 pub fn get_linked_version() -> Version {
-    unsafe { Version::from_ll(*ttf::TTF_Linked_Version()) }
-}
-
-/// An error for when `sdl2_ttf` is attempted initialized twice
-/// Necessary for context management, unless we find a way to have a singleton
-#[derive(Debug)]
-pub enum InitError {
-    InitializationError(io::Error),
-    AlreadyInitializedError,
-}
-
-impl error::Error for InitError {
-    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
-        match *self {
-            InitError::AlreadyInitializedError => None,
-            InitError::InitializationError(ref error) => Some(error),
-        }
-    }
-}
-
-impl fmt::Display for InitError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Self::AlreadyInitializedError => {
-                write!(f, "SDL2_TTF has already been initialized")
-            }
-            Self::InitializationError(error) => write!(f, "SDL2_TTF initialization error: {error}"),
-        }
-    }
+    Version::from_ll(unsafe { *ttf::TTF_Linked_Version() })
 }
 
 /// Initializes the truetype font API and returns a context manager which will
 /// clean up the library once it goes out of scope.
-pub fn init() -> Result<Sdl2TtfContext, InitError> {
-    unsafe {
-        if ttf::TTF_WasInit() == 1 {
-            Err(InitError::AlreadyInitializedError)
-        } else if ttf::TTF_Init() == 0 {
-            Ok(Sdl2TtfContext)
-        } else {
-            Err(InitError::InitializationError(io::Error::last_os_error()))
-        }
+#[doc(alias = "TTF_Init")]
+pub fn init() -> Result<Sdl2TtfContext, String> {
```

Comment:
I changed the error type to `String` here (like in other functions in this library that only return an error from sdl directly), but this specifically will break in an annoying way because now the error does not implement `std::error::Error` anymore (relevant: #942, #1053).

I feel like introducing a struct that wraps `String` (like `pub struct SDLError(String);`) and implements `std::error::Error` and `Display`. This will (in the future) also allow other enums in this crate that can contain an SDL error string to return it from `std::error::Error::source`.